src/main/java/pl/put/poznan/buildingInfo/model/Janitor.java

```diff
@@ -83,4 +83,11 @@ public static boolean compareLocationID(Location l1, Location l2, int i) {
         return (ArrayList) (level.getRooms().stream()
             .filter(room -> room.getHeating()/room.getCube() > limit).collect(Collectors.toList()));
     }
+
+    public static boolean verifyUniqueBuildingId(Building building, List<Building> buildings) {
+        String buildingId = building.getId();
+        if (buildings.stream()
```

Comment:
I'm pretty sure that's what distinctByKey does but I guess you could do without using predicates :P

Reply:
After looking through it, I came to conclusion that it's not really the same. Both methods do their own work.

Reply:
Not sure if I can, I'm comparing new building's id, with other buildings id. They're not in the same list. If it's intended to use, I can add newly created building to lsit and the go with distinctByKey

Reply:
I think what just happend could be described ad 'Phantom read' :P

Reply:
yea 😄 so then, should I go like that?
``` 
long before = buildings.stream().filter(distinctByKey(b -> b.getId())).count();
buildings.add(building);
return buildings.stream().filter(distinctByKey(b -> b.getId())).count() == before;
```

Reply:
Both ways work. I'm pretty sure though what you just wrote would make my metod feel more useful :D

Reply:
Let it be then! 😃